### PR TITLE
[CK_TILE][FMHA] Enable wholek_prefetch

### PR DESCRIPTION
**Pull request by LJ-underdog**
## Proposed changes

Please describe the motivation behind the pull request, whether it enables a new feature or fixes a bug. If there are associated pull requests or issues, please link them to the pull request.

## Checklist

Please put an `x` into the boxes that apply. You can also fill these out after creating the PR. If you're not sure, please don't hesitate to ask.

- [ ] I have added tests relevant to the introduced functionality, and the unit tests are passing locally
- [ ] I have added the test to REGRESSION_TESTS list defined at the top of CMakeLists.txt in tests/CMakeLists.txt, **IF** the test takes more than 30 seconds to run.
- [ ] I have added inline documentation which enables the maintainers with understanding the motivation
- [ ] I have removed the stale documentation which is no longer relevant after this pull request
- [ ] (If this change is user-facing) I have added release notes which provide the end users with a brief summary of the improvement from this pull request
- [ ] I have run `clang-format` on all changed files
- [ ] Any dependent changes have been merged

## Discussion

If this is a relatively large or complex change, feel free to start a discussion by explaining why you chose the solution you did and what alternatives you considered

### CODE REVIEW

**Comment by poyenc** [comment]
@LJ-underdog are you still working on this?

**Comment by LJ-underdog** [comment]
> @LJ-underdog are you still working on this?

yes， this pr is prepare for mate

File: example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py [hunk 1]
```diff
@@ -1320,6 +1359,10 @@ def get_fwd_blobs(
             for tile, pipeline in itertools.product(
                 tiles, factory.get_pipelines(dtype, hdim, hdim_v, receipt, mask_impl)
             ):
+                if pipeline.tag == "qr_wholek_prefetch" and (
```
Comment (poyenc) [inline example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py:1364]: Please move the check to the `CompatibilityRuleFactoryGfx9`

File: example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py [hunk 2]
```diff
@@ -1012,6 +1027,32 @@ def get_pipelines(
...
+                            and logits == "f"
+                            and bias == "no"
+                            and dropout == "f"
+                            and sink == "f"
+                        ):
+                            pipelines.append(
+                                FmhaFwdPipeline(
+                                    "qr_wholek_prefetch",
+                                    "row",
+                                    "f",
```
Comment (poyenc) [inline example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py:1041]: Why don't add padding for both of the `seqlen_q` & `seqlen_k` dimensions?

**Comment by illsilin** [comment]
Hi @LJ-underdog , please make sure this thing compiles and passes tests!
